query-node/mappings/src/content/video.ts

```diff
@@ -246,6 +267,87 @@ export async function content_VideoDeleted({ store, event }: EventContext & Stor
+  // get referencing VideoReaction
+  const referencingVideoReactions = await loadReferencingEntities(store, VideoReaction, videoId)
+  // get referencing VideoReactionsCountByReactionType
+  const referencingVideoReactionsCountByReactionTypes = await loadReferencingEntities(
+    store,
+    VideoReactionsCountByReactionType,
+    videoId
+  )
+  // get referencing CommentReactionsCountByReactionId
+  const referencingCommentReactionsCountByReactionIds = await loadReferencingEntities(
+    store,
+    CommentReactionsCountByReactionId,
+    videoId
+  )
+  // get referencing Comments
+  const referencingComments = await loadReferencingEntities(store, Comment, videoId)
+
+  /**
+   * Referencing Event Entities
+   */
+  // referencing VideoReactedEvent
+  const referencingVideoReactedEvents = await loadReferencingEntities(store, VideoReactedEvent, videoId)
+  // referencing CommentReactedEvent
+  const referencingCommentReactedEvents = await loadReferencingEntities(store, CommentReactedEvent, videoId)
+  // referencing CommentCreatedEvent
+  const referencingCommentCreatedEvents = await loadReferencingEntities(store, CommentCreatedEvent, videoId)
+  // referencing CommentTextUpdatedEvent
+  const referencingCommentTextUpdatedEvents = await loadReferencingEntities(store, CommentTextUpdatedEvent, videoId)
+  // referencing CommentDeletedEvent
+  const referencingCommentDeletedEvents = await loadReferencingEntities(store, CommentDeletedEvent, videoId)
+  // referencing CommentModeratedEvent
+  const referencingCommentModeratedEvents = await loadReferencingEntities(store, CommentModeratedEvent, videoId)
+  // referencing CommentPinnedEvent
+  const referencingCommentPinnedEvents = await loadReferencingEntities(store, CommentPinnedEvent, videoId)
+  // referencing VideoReactionsPreferenceEvent
+  const referencingVideoReactionsPreferenceEvents = await loadReferencingEntities(
+    store,
+    VideoReactionsPreferenceEvent,
+    videoId
+  )
+
+  const removeRelations = async <T>(store: DatabaseManager, entities: T[]) => {
+    await Promise.all(entities.map(async (r) => await store.remove<T>(r)))
+  }
+
+  // remove referencing records
+  await removeRelations(store, referencingCommentReactions)
+  await removeRelations(store, referencingVideoReactions)
+  await removeRelations(store, referencingVideoReactionsCountByReactionTypes)
+  await removeRelations(store, referencingCommentReactionsCountByReactionIds)
+  await removeRelations(store, referencingVideoReactedEvents)
+  await removeRelations(store, referencingCommentReactedEvents)
+  await removeRelations(store, referencingCommentCreatedEvents)
+  await removeRelations(store, referencingCommentTextUpdatedEvents)
+  await removeRelations(store, referencingCommentDeletedEvents)
+  await removeRelations(store, referencingCommentModeratedEvents)
+  await removeRelations(store, referencingCommentPinnedEvents)
+  await removeRelations(store, referencingVideoReactionsPreferenceEvents)
```

Comment:
There are a bunch of similar lines. Let's create an array (`[VideoReactedEvent, CommentReactedEvent, ...]`) and iterate over it rather than have a series of `loadReferencingEntities` and `removeRelations` calls.